packages/assistify-dbs-translation/server/dbsTranslate.js

```diff
@@ -0,0 +1,218 @@
+	 * @param {string} target
+	 * @returns {object} code : value pair
+	 */
+	_getSupportedLanguages(target) {
+		if (this.autoTranslateEnabled && this.apiKey) {
+			if (this.supportedLanguages[target]) {
+				return this.supportedLanguages[target];
+			}
+			return this.supportedLanguages[target] = [
+				{
+					'language': 'de',
+					'name': TAPi18n.__('German', {lng: target})
+				},
+				{
+					'language': 'en',
+					'name': TAPi18n.__('English', {lng: target})
+				},
+				{
+					'language': 'fr',
+					'name': TAPi18n.__('French', {lng: target})
+				},
+				{
+					'language': 'es',
+					'name': TAPi18n.__('Spanish', {lng: target})
+				},
+				{
+					'language': 'it',
+					'name': TAPi18n.__('Italian', {lng: target})
+				},
+				{
+					'language': 'nl',
+					'name': TAPi18n.__('Dutch', {lng: target})
+				},
+				{
+					'language': 'pl',
+					'name': TAPi18n.__('Polish', {lng: target})
+				},
+				{
+					'language': 'ro',
+					'name': TAPi18n.__('Romanian', {lng: target})
+				},
+				{
+					'language': 'sk',
+					'name': TAPi18n.__('Slovak', {lng: target})
+				},
+				{
+					'language': 'ja',
+					'name': TAPi18n.__('Japanese', {lng: target})
+				},
+				{
+					'language': 'zh',
+					'name': TAPi18n.__('Chinese', {lng: target})
+				}
+			];
+		}
+	}
+
+	/**
+	 * Send Request REST API call to the service provider.
+	 * Returns translated message for each target language in target languages.
+	 * @private
+	 * @param {object} message
+	 * @param {object} targetLanguages
+	 * @returns {object} translations: Translated messages for each language
+	 */
+	_sendRequestTranslateMessage(message, targetLanguages) {
+		const translations = {};
+		const msgs = message.msg.split('\n');
+		const query = msgs.join();
+		let sourceLanguage;
+		/**
+		 * Service provider do not handle the text language detection automatically rather it requires the source language to be specified
+		 * explicitly. To automate this language detection process we used the cld language detector.
+		 * When the language detector fails, it falls back into the user language.
+		 */
+		cld.detect(query, (err, result) => {
+			if (result) {
+				result.languages.map((language) => {
+					sourceLanguage = language.code;
+				});
+				const supportedLanguages = this._getSupportedLanguages('en');
+				targetLanguages.forEach(language => {
+					if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, {language})) {
+						language = language.substr(0, 2);
+					}
+					const result = Meteor.call('assistify.translate', 'POST', `${ this.apiEndPointUrl }/translate`, {
+						params: {
+							key: this.apiKey
+						}, data: {
+							text: query,
+							to: language,
+							from: sourceLanguage
+						}
+					});
+					translations[language] = this.deTokenize(Object.assign({}, message, {msg: result}));
+				});
+			} else {
+				SystemLogger.log('Text language could not be determined', err.message);
+			}
+		});
+		return translations;
```

Comment:
this return is outside the callback => should this be `await`ed (return a Promise)?